examples/http_loader.rb

```diff
@@ -0,0 +1,64 @@
+# A sample HTTP loader using:
+#
+# 1. https://github.com/httprb/http
+# 2. https://github.com/mperham/connection_pool
+#
+# Setup:
+#
+#   field :weather, String, null: true do
+#     argument :lat, Float, required: true
+#     argument :lng, Float, required: true
+#     argument :lang, String, required: true
+#   end
+#
+#   def weather(lat:, lng:, lang:)
+#     key = Rails.application.credentials.darksky_key
+#     path = "/forecast/#{key}/#{lat},#{lng}?lang=#{lang}"
+#     Loaders::HTTPLoader
+#       .for(host: 'https://api.darksky.net')
+#       .load(->(connection) { connection.get(path).flush })
+#       .then do |response|
+#         if response.status.ok?
+#           json = JSON.parse(response.body)
+#           json['currently']['summary']
+#         end
+#       end
+#   end
+#
+# Querying:
+#
+#   <<~GQL
+#     query Weather {
+#       montreal: weather(lat: 45.5017, lng: -73.5673, lang: "fr")
+#       waterloo: weather(lat: 43.4643, lng: -80.5204, lang: "en")
+#     }
+#   GQL
+
+module Loaders
+  class HTTPLoader < GraphQL::Batch::Loader
+    def initialize(host:, size: 4, timeout: 4)
+      @host = host
+      @size = size
+      @timeout = timeout
+    end
+
+    def perform(operations)
+      threads = operations.map do |operation|
+        Thread.new do
```

Comment:
This is great, thank you!

Just a quick question: it looks like `ConnectionPool` will correctly keep some number of initialized HTTP clients ready to perform requests. But is there a problem here if the number of `operations` is arbitrarily high, and hence we could spawn an arbitrary number of threads?

I wonder if https://github.com/ruby-concurrency/concurrent-ruby/blob/master/docs-source/thread_pools.md is also useful here?

Reply:
@chrisbutcher good callout - I think it'd be fairly easy to exhaust the list (if some sort of complexity rules aren't applied):

https://graphql-ruby.org/queries/complexity_and_depth#prevent-complex-queries

I'll make some alterations tonight.

Reply:
@chrisbutcher looking through the docs for thread pool within concurrent-ruby it looks like it'd require using a mutex to guard a critical section for saving results into a collection. 

Instead of taking that approach - any objection to using the higher-order promise concurrency paradigm?: 

https://github.com/Shopify/graphql-batch/pull/113/commits/23f8740556ff2189e4f222c4be02c0e0e81a9de0

Reply:
Looks great to me. Thanks Kevin!